libsolidity/formal/SMTChecker.cpp

```diff
@@ -736,9 +736,9 @@ bool SMTChecker::visit(MemberAccess const& _memberAccess)
 
```

Comment:
Also the above is weird: What if you have code like `f(<some side-effecty code>).g` where `f` returns a contract and `g` is one of the contract's functinos?

Reply:
Then the function call would be inlined if possible and the member access would have no effect.

Reply:
Ah sorry - `return true`

Reply:
But still, I think it is very dangerous to perform checks of the resulting type and not the type of the expression and the name of the member.

Reply:
But the code also checks that the expression is an `EnumDefinition`

Reply:
In any case, I see the point, will change.